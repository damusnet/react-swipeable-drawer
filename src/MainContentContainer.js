import React from "react";
import PropTypes from "prop-types";

const MainContentContainer = ({ translation, mainContentScroll, children }) => {
  const mainContentOpenStyle =
    translation > 0
      ? {
          position: "fixed",
          top: -mainContentScroll,
          left: 0,
          right: 0,
        }
      : {};

  return (
    <div className="MainContentContainer" style={{ ...mainContentOpenStyle }}>
      {children}
    </div>
  );
};

export default MainContentContainer;

MainContentContainer.propTypes = {
  translation: PropTypes.number.isRequired,
  mainContentScroll: PropTypes.number.isRequired,
  children: PropTypes.node.isRequired,
};
