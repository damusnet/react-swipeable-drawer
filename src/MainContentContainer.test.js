import React from "react";
import renderer from "react-test-renderer";

import MainContentContainer from "./MainContentContainer";

describe("<MainContentContainer />", () => {
  it("renders correctly when closed", () => {
    const component = renderer.create(
      <MainContentContainer translateX={0} mainContentScroll={0}>
        <div />
      </MainContentContainer>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when swiping", () => {
    const component = renderer.create(
      <MainContentContainer translateX={50} mainContentScroll={100}>
        <div />
      </MainContentContainer>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
