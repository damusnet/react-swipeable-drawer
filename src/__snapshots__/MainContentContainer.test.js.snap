// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContentContainer /> renders correctly when closed 1`] = `
<div
  className="MainContentContainer"
  style={Object {}}
>
  <div />
</div>
`;

exports[`<MainContentContainer /> renders correctly when swiping 1`] = `
<div
  className="MainContentContainer"
  style={
    Object {
      "position": "fixed",
      "top": -100,
    }
  }
>
  <div />
</div>
`;
